src/FormField/ScopeBuilder.php

```diff
@@ -0,0 +1,576 @@
+<?php
+
+namespace atk4\ui\FormField;
+
+use atk4\data\Model;
+use atk4\ui\Exception;
+use atk4\ui\Template;
+use atk4\data\Field;
+use atk4\data\Model\Scope\Condition;
+use atk4\data\Model\Scope\Scope;
+use atk4\data\Model\Scope\AbstractScope;
+
+class ScopeBuilder extends Generic
+{
+    /**
+     * Field type specific options.
+     *
+     * @var array
+     */
+    public $options = [
+        'enum' => [
+            'limit' => 250
+        ]
+    ];
+    /**
+     * Max depth of nested conditions allowed.
+     * Corresponds to VueQueryBulder maxDepth
+     *
+     * @var integer
+     */
+    public $maxDepth = 10;
+
+    /**
+     * Fields to use for creating the rules
+     *
+     * @var array
+     */
+    public $fields = [];
+
+    /**
+     * The template needed for the scopebuilder view.
+     *
+     * @var Template
+     */
+    public $scopeBuilderTemplate;
+
+    /**
+     * List of delimiters for auto-detection in order of priority
+     *
+     * @var array
+     */
+    public static $listDelimiters = [';', ',', '|'];
+
+    /**
+     * The scopebuilder View. Assigned in init().
+     *
+     * @var \atk4\ui\View
+     */
+    protected $scopeBuilderView;
+
+    /**
+     * Definition of VueQueryBuilder rules.
+     *
+     * @var array
+     */
+    protected $rules = [];
+
+    /**
+     * Default VueQueryBuilder query.
+     *
+     * @var array
+     */
+    protected $query = [];
+
+    /**
+     * VueQueryBulder => Condition map of operators
+     *
+     * @var array
+     */
+    protected static $operators = [
+        'equals' => '=',
```

Comment:
@georgehristov you use these natural strings on multiple locations, can you declare them in constants once and then use them?

Example:
```
public const OPERATOR_DOES_NOT_EQUAL = 'does not equal';

something(self::OPERATOR_DOES_NOT_EQUAL);
```

Reply:
@mvorisek I also like using constants for definition but the challenge here is that we have operator label and operator value. Will think about how to implement it.

Reply:
![image](https://user-images.githubusercontent.com/2228672/84356754-f0327600-abc4-11ea-8548-c8021feaa8e5.png)

something that is unique... it needs to be probably the name/label :)